ecl/eclagent/eclagent.cpp

```diff
@@ -2075,14 +2059,14 @@ void EclAgent::doProcess()
         logException((IException *) NULL);
     }
     try {
-        unlockWorkUnit();
+        w->commit();
     }
     catch (IException *e)
     {
         try
         {
             // a final attempt to commit the original error (only) to the workunit
-            // since unlockWorkUnit( which commits ) can error due to the nature of the transaction.
+            // since commit () can error due to the nature of the transaction.
```

Comment:
I'm not if this is a real problem, but before releasing connection and failing during the implicit commit(), would still release the lock.

But now, if w->commit() on line 2062 throws an exception, w will still hold the lock to the workunit.
that should mean the code below deadlocks trying to get a 2nd exclusive lock on the workunit.

I think it would be ok, if the line on 2062 was : w.clear();

Reply:
NB: I also suspect this code should restablish the write lock with: 
```
w.setown(updateWorkUnit());
```

.. rather than factory again, but I wouldn't change it, without further investigation, e.g. reproduce a fake commit failure etc.